ls1intum/Artemis
# Feature/restore automatic feedback

### Checklist

### Motivation and Context
- Currently tutors don't have an opportunity to restore the original automatic feedback once they changed it. This PR brings this possibility to text-exercises.
- For my Bachelor-Thesis, I need to store the original automatic feedback for text exercises in order to analyze the modifications made by the tutors.

### Description
- Added a new 'automatic_detail_text' column to the data model (Client, Server & Database)
- Add a restore icon to the text exercise assessment
- Modify the `feedback.model.ts` to handle changes back to the automatic feedback

### Steps for Testing on Server

1. Deploy to Test Server 1
2. Login to Test Server 1
3. Open [this](https://artemistest.ase.in.tum.de/#/course-management/249/text-exercises/2215/submissions/303559/assessment) assessment
4. See the "automatic" badge on the feedback comment
5. Change the feedback comment
6. See "restore" button
7. Click & confirm the "restore" button
8. See the original comment and the automatic badge again

### Steps for Testing locally


1. Log in to Artemis
2. Navigate to Course Administration
3. Find a text exercise that has an assessed solution
3.1 Alternative: Create the assessed solution
3.2 Create a text exercise
3.3 Submit a participation for this exercise
3.4 Review the exercise
3.5 Submit your feedback
4. Go into you local database and set one of the feedback comments of the assessment above to type = `AUTOMATIC`
5. Re-Assess the exercise
6. See the "automatic" badge on the feedback you just changed
7. Change this feedback comment
8. See "restore" button
9. Click & confirm the "restore" button
10. See the original comment and the automatic badge again

### Screenshots



## Automatic feedback is displayed - nothing changed here
![image](https://user-images.githubusercontent.com/51994246/83949951-47cc8c80-a827-11ea-87a3-0c693ded0d9f.png)

## Automatic feedback is changed
![image](https://user-images.githubusercontent.com/51994246/83949963-67fc4b80-a827-11ea-9e41-c80396ee3001.png)
